README.md
DOSE: Disease Ontology Semantic and Enrichment analysis
=======================================================

[![](https://img.shields.io/badge/release%20version-3.4.0-green.svg)](https://bioconductor.org/packages/DOSE) [![](https://img.shields.io/badge/devel%20version-3.5.1-green.svg)](https://github.com/guangchuangyu/DOSE) [![Bioc](http://www.bioconductor.org/shields/years-in-bioc/DOSE.svg)](https://www.bioconductor.org/packages/devel/bioc/html/DOSE.html#since)

[![platform](http://www.bioconductor.org/shields/availability/devel/DOSE.svg)](https://www.bioconductor.org/packages/devel/bioc/html/DOSE.html#archives) [![Build Status](http://www.bioconductor.org/shields/build/devel/bioc/DOSE.svg)](https://bioconductor.org/checkResults/devel/bioc-LATEST/DOSE/) [![Linux/Mac Travis Build Status](https://img.shields.io/travis/GuangchuangYu/DOSE/master.svg?label=Mac%20OSX%20%26%20Linux)](https://travis-ci.org/GuangchuangYu/DOSE) [![AppVeyor Build Status](https://img.shields.io/appveyor/ci/Guangchuangyu/DOSE/master.svg?label=Windows)](https://ci.appveyor.com/project/GuangchuangYu/DOSE)

[![Project Status: Active - The project has reached a stable, usable state and is being actively developed.](http://www.repostatus.org/badges/latest/active.svg)](http://www.repostatus.org/#active) [![codecov](https://codecov.io/gh/GuangchuangYu/DOSE/branch/master/graph/badge.svg)](https://codecov.io/gh/GuangchuangYu/DOSE/) [![Last-changedate](https://img.shields.io/badge/last%20change-2018--04--09-green.svg)](https://github.com/GuangchuangYu/DOSE/commits/master)

This package implements five methods proposed by *Resnik*, *Schlicker*, *Jiang*, *Lin* and *Wang* respectively for measuring semantic similarities among DO terms and gene products. Enrichment analyses including hypergeometric model and gene set enrichment analysis are also implemented for discovering disease associations of high-throughput biological data.

For details, please visit our project website, <https://guangchuangyu.github.io/software/DOSE>.

-   [Documentation](https://guangchuangyu.github.io/software/DOSE/documentation/)
-   [Featured Articles](https://guangchuangyu.github.io/software/DOSE/featuredArticles/)
-   [Feedback](https://guangchuangyu.github.io/software/DOSE/#feedback)

[![Twitter](https://img.shields.io/twitter/url/http/shields.io.svg?style=social&logo=twitter)](https://twitter.com/intent/tweet?hashtags=DOSE&url=http://bioinformatics.oxfordjournals.org/content/31/4/608) [![saythanks](https://img.shields.io/badge/say-thanks-ff69b4.svg)](https://saythanks.io/to/GuangchuangYu) [![](https://img.shields.io/badge/follow%20me%20on-微信-green.svg)](https://guangchuangyu.github.io/blog_images/biobabble.jpg) [![](https://img.shields.io/badge/打赏-支付宝/微信-green.svg)](https://guangchuangyu.github.io/blog_images/pay_qrcode.png)

------------------------------------------------------------------------

Please cite the following article when using `DOSE`:

***G Yu***, LG Wang, GR Yan, QY He. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis. ***Bioinformatics*** 2015, 31(4):608-609.

[![](https://img.shields.io/badge/doi-10.1093/bioinformatics/btu684-green.svg)](http://dx.doi.org/10.1093/bioinformatics/btu684) [![](https://img.shields.io/badge/Altmetric-26-green.svg)](https://www.altmetric.com/details/2788597)

------------------------------------------------------------------------

### Citation

<img src="https://guangchuangyu.github.io/software/DOSE/featured-articles/index_files/figure-html/citation-1.png" width="890"/>

### Download stats

[![download](http://www.bioconductor.org/shields/downloads/DOSE.svg)](https://bioconductor.org/packages/stats/bioc/DOSE) [![](https://img.shields.io/badge/download-65637/total-blue.svg)](https://bioconductor.org/packages/stats/bioc/DOSE) [![](https://img.shields.io/badge/download-2907/month-blue.svg)](https://bioconductor.org/packages/stats/bioc/DOSE)

<img src="https://guangchuangyu.github.io/software/DOSE/index_files/figure-html/dlstats-1.png" width="890"/>


NEWS
CHANGES IN VERSION 3.5.2
------------------------
 o bug fixed of gseaScores <2018-04-18, Wed>
   + https://github.com/GuangchuangYu/DOSE/issues/23
 o mv web site to https://guangchuangyu.github.io/software/DOSE

CHANGES IN VERSION 3.3.2
------------------------
 o new project site using blogdown <2017-09-28, Thu>
 o ridgeplot for gseaResult <2017-08-17, Thu>

CHANGES IN VERSION 3.3.1
------------------------
 o throw warning instead of error when fail to `setReadable`. <2017-06-28, Wed>
   + https://github.com/GuangchuangYu/clusterProfiler/issues/91
 o better msg when using wrong ID types in GSEA <2017-06-01, Thu>
   + https://support.bioconductor.org/p/96512/#96516

CHANGES IN VERSION 3.2.0
------------------------
 o BioC 3.5 release <2017-04-26, Wed>

CHANGES IN VERSION 3.1.3
------------------------
 o output expected sample gene ID when input gene ID type not match <2017-03-27, Mon>
 o dotplot for gseaResult <2016-11-23, Fri>
   + https://github.com/GuangchuangYu/DOSE/issues/20

CHANGES IN VERSION 3.1.2
------------------------
 o in gseaplot, call grid.newpage only if dev.interactive() <2016-11-16, Wed>
 o change minGSSize <  geneSet_size & geneSet_size <  maxGSSize to
   	  minGSSize <= geneSet_size & geneSet_size <= maxGSSize <2016-11-16, Wed>
 o fixed show method issus of unknown setType in clusterProfiler::GSEA output <2016-11-15, Tue>
 o throw more friendly error msg if fail to determine setType automatically in setReadable function <2016-11-15, Tue>
   + https://support.bioconductor.org/p/89445/#89479
 o apply minGSSize and maxGSSize to fgsea <2016-11-14, Mon>
 o change summary to as.data.frame in internal calls to prevent warning message <2016-11-14, Mon>

CHANGES IN VERSION 3.1.1
------------------------
 o update startup message <2016-11-09, Wed>
 o fixed parallel in Windows (not supported) <2016-10-24, Mon>
   + https://github.com/GuangchuangYu/DOSE/issues/16
 o options(DOSE_workers = x) to set using x cores for GSEA analysis is removed. <2016-10-24, Mon>
   instead let MulticoreParam() to decide how many cores (can be set by `options(mc.cores = x)`).

CHANGES IN VERSION 3.0.0
------------------------
 o BioC 3.4 released <2016-10-18, Tue>

CHANGES IN VERSION 2.11.12
------------------------
 o deprecate summary method and use as.data.frame instead <2016-09-29, Thu>

CHANGES IN VERSION 2.11.11
------------------------
 o export geneID and geneInCategory <2016-09-19, Mon>

CHANGES IN VERSION 2.11.10
------------------------
 o geneID and geneInCategory accessor functions <2016-09-19, Mon>

CHANGES IN VERSION 2.11.9
------------------------
 o enrichMap works with result contains only one term <2016-08-15, Wed>
   + https://github.com/GuangchuangYu/DOSE/issues/15
 o build_Anno now works with tibble <2016-08-18, Tue>

CHANGES IN VERSION 2.11.8
------------------------
 o add unit test <2016-08-15, Mon>

CHANGES IN VERSION 2.11.7
------------------------
 o change for meshes packages <2016-08-11, Thu>

CHANGES IN VERSION 2.11.6
------------------------
 o user can use options(DOSE_workers = x) to set using x cores for GSEA analysis <2016-08-02, Tue>
 o support DisGeNET enrichment analyses <2016-08-01, Mon>
   + enrichDGN, enrichDGNv, gseDGN
 o update vignettes <2016-07-29, Fri>

CHANGES IN VERSION 2.11.5
------------------------
 o enrichMap now output igraph object that can be viewed using other software like networkD3 <2016-07-25, Mon>
 o dim methods for enrichResult and gseaResult <2016-07-22, Fri>
 o $ methods for enrichResult and gseaResult <2016-07-20, Wed>
 o switch from parallel to BiocParallel <2017-07-07, Thu>
 o [, head and tail methods for enrichResult and gseaResult <2016-07-06, Wed>
 o change according to GOSemSim <2016-07-05, Tue>

CHANGES IN VERSION 2.11.4
------------------------
 o 'title' parameter for gseaplot <2016-07-04, Mon>
   + contributed by https://github.com/pedrostrusso
   + https://github.com/GuangchuangYu/DOSE/pull/13
 o 'by' parameter in GSEA_internal, by default by = 'fgsea' <2016-07-04, Mon>
   + by = 'fgsea', use GSEA algorithm implemented in fgsea
   + by = 'DOSE', use GSEA algorithm implemented in DOSE
 o leading edge analysis for GSEA <2016-07-04, Mon>

CHANGES IN VERSION 2.11.3
------------------------
 o output igraph object in cnetplot <2016-06-21, Tue>
 o upsetplot generics <2016-06-14, Tue>
 o [[ methods for enrichResult and gseaResult for accessing gene set <2016-06-14, Tue>

CHANGES IN VERSION 2.11.2
------------------------
 o use byte compiler <2016-05-18, Wed>
 o https://github.com/Bioconductor-mirror/DOSE/commit/6c508c6a6816f465bb372f30f4ab99c839d81767

CHANGES IN VERSION 2.11.1
------------------------
 o https://github.com/Bioconductor-mirror/DOSE/commit/7e87d01e671ce1b5fbe974c06b796b1a2970f11c

CHANGES IN VERSION 2.10.0
------------------------
 o BioC 3.3 released <2016-05-05, Thu>

CHANGES IN VERSION 2.9.7
------------------------
 o barplot accepts x and colorBy parameters as in dotplot <2016-04-13, Wed>
 o gsfilter function for restricting result with minimal and maximal gene set size <2016-03-31, Thu>
   + see https://github.com/GuangchuangYu/clusterProfiler/issues/46

CHANGES IN VERSION 2.9.6
------------------------
 o add maxGSSize parameter for hypergeometric test <2016-03-09, Wed>
 o add maxGSSize parameter in GSEA analysis. <2016-03-09, Wed>
   + Default of 500 is fairly standard for GSEA analysis.
   + Usually if the geneset > 500, its probability of being called significant by GSEA rises quite dramatically.
 o fixed R check <2016-03-05, Sat>
 o update ReactomePA citation info <2016-02-17, Wed>

CHANGES IN VERSION 2.9.5
------------------------
 o upset plot for enrichResult object <2016-01-22, Fri>

CHANGES IN VERSION 2.9.4
------------------------
 o bug fixed in scaling category sizes in enrichMap <2016-01-10, Sun>
   + use setSize in gseResult

CHANGES IN VERSION 2.9.3
------------------------
 o update enrichMap to scale category sizes <2016-01-04, Mon>

R/print.R
##' show method for \code{gseaResult} instance
##'
##' @name show
##' @docType methods
##' @rdname show-methods
##'
##' @title show method
##' @return message
##' @importFrom methods show
##' @exportMethod show
##' @usage show(object)
##' @author Guangchuang Yu \url{https://guangchuangyu.github.io}
setMethod("show", signature(object="gseaResult"),
          function (object){
              params <- object@params
              cat("#\n# Gene Set Enrichment Analysis\n#\n")
              cat("#...@organism", "\t", object@organism, "\n")
              cat("#...@setType", "\t", object@setType, "\n")
              kt <- object@keytype
              if (kt != "UNKNOWN") {
                  cat("#...@keytype", "\t", kt, "\n")
              }

              cat("#...@geneList", "\t")
              str(object@geneList)
              cat("#...nPerm", "\t", params$nPerm, "\n")
              cat("#...pvalues adjusted by", paste0("'", params$pAdjustMethod, "'"),
                  paste0("with cutoff <", params$pvalueCutoff), "\n")
              cat(paste0("#...", nrow(object@result)), "enriched terms found\n")
              str(object@result)
              cat("#...Citation\n")
              if (object@setType == "DO" || object@setType == "DOLite" || object@setType == "NCG") {
                  citation_msg <- paste("  Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an",
                                    "  R/Bioconductor package for Disease Ontology Semantic and Enrichment",
                                    "  analysis. Bioinformatics 2015, 31(4):608-609", sep="\n", collapse="\n")
              } else if (object@setType == "Reactome") {
                  citation_msg <- paste("  Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for",
                                        "  reactome pathway analysis and visualization. Molecular BioSystems",
                                        "  2016, 12(2):477-479", sep="\n", collapse="\n")
              } else {
                  citation_msg <- paste("  Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.",
                                        "  clusterProfiler: an R package for comparing biological themes among",
                                        "  gene clusters. OMICS: A Journal of Integrative Biology",
                                        "  2012, 16(5):284-287", sep="\n", collapse="\n")
              }
              cat(citation_msg, "\n\n")
          }
          )

##' show method for \code{enrichResult} instance
##'
##' @name show
##' @docType methods
##' @rdname show-methods
##'
##' @title show method
##' @param object A \code{enrichResult} instance.
##' @return message
##' @importFrom utils str
##' @importFrom methods show
##' @exportMethod show
##' @usage show(object)
##' @author Guangchuang Yu \url{https://guangchuangyu.github.io}
setMethod("show", signature(object="enrichResult"),
          function (object){
              cat("#\n# over-representation test\n#\n")
              cat("#...@organism", "\t", object@organism, "\n")
              cat("#...@ontology", "\t", object@ontology, "\n")
              kt <- object@keytype
              if (kt != "UNKNOWN") {
                  cat("#...@keytype", "\t", kt, "\n")
              }
              cat("#...@gene", "\t")
              str(object@gene)
              cat("#...pvalues adjusted by", paste0("'", object@pAdjustMethod, "'"),
                  paste0("with cutoff <", object@pvalueCutoff), "\n")
              object <- get_enriched(object)
              n <- nrow(object@result)
              cat(paste0("#...", n), "enriched terms found\n")
              if (n > 0) str(object@result)
              cat("#...Citation\n")
              if (object@ontology == "DO" || object@ontology == "DOLite" || object@ontology == "NCG") {
                  citation_msg <- paste("  Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an",
                                    "  R/Bioconductor package for Disease Ontology Semantic and Enrichment",
                                    "  analysis. Bioinformatics 2015, 31(4):608-609", sep="\n", collapse="\n")
              } else if (object@ontology == "Reactome") {
                  citation_msg <- paste("  Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for",
                                        "  reactome pathway analysis and visualization. Molecular BioSystems",
                                        "  2016, 12(2):477-479", sep="\n", collapse="\n")
              } else {
                  citation_msg <- paste("  Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.",
                                        "  clusterProfiler: an R package for comparing biological themes among",
                                        "  gene clusters. OMICS: A Journal of Integrative Biology",
                                        "  2012, 16(5):284-287", sep="\n", collapse="\n")
              }
              cat(citation_msg, "\n\n")
          })
